### PR TITLE
Update bots to use SUBSPACE_BUILD_CIR option.

### PR DESCRIPTION
**Pull request by dj2**
This CL converts the bots from using the `SUBSPACE_NO_BUILD_CIR` environment variable to the new cmake `SUBSPACE_BUILD_CIR` option.

### CODE REVIEW

**Comment by dj2** [comment]
I don't know how to test this ..... hopefully it works?

**Comment by danakj** [comment]
Yeah theres no good way. I will land it and watch it tomorrow, fix anything if it goes wrong. Thank you!

File: .github/workflows/clang-doc.yml [hunk 1]
```diff
@@ -24,17 +24,13 @@ jobs:
           sudo ./llvm.sh 16 all
 
       - name: Configure
-        env:
-          # Skip building documentation for Cir, which ends up building docs for all of
```
Comment (danakj) [inline .github/workflows/clang-doc.yml:28]: Save the comment?

**Review by danakj — COMMENTED** [review]
Thanks! 1 request

File: .github/workflows/clang-doc.yml (hunk 1 above)
Comment (dj2) [reply .github/workflows/clang-doc.yml:28]: done.

**Comment by danakj** [comment]
Thank you =)

**Comment by danakj** [comment]
FYI msvc and docs bots passed, so I think this was bug free.

**Pull request merged**